src/index.js

```diff
@@ -193,19 +202,47 @@ class IOpipeWrapperClass {
       this.originalContext.done(err, data);
     });
   }
-  log(name, value) {
+  metric(keyInput, valueInput) {
     let numberValue, stringValue;
-    if (typeof value === 'number') {
-      numberValue = value;
+    const key = convertToString(keyInput);
+    if (key.length > 128) {
+      this.debugLog(
+        `Metric with key name ${key} is longer than allowed length of 128, metric will not be saved`
+      );
+      return;
+    }
+    if (Number.isFinite(valueInput)) {
+      numberValue = valueInput;
     } else {
-      stringValue = typeof value === 'string' ? value : JSON.stringify(value);
+      stringValue = convertToString(valueInput);
     }
     this.metrics.push({
-      name,
+      name: key,
       n: numberValue,
       s: stringValue
     });
   }
+  label(name) {
+    if (typeof name !== 'string') {
+      this.debugLog(`Label ${name} is not a string and will not be saved`);
+      return;
+    }
+    if (name.length > 128) {
+      this.debugLog(
+        `Label with name ${name} is longer than allowed length of 128, label will not be saved`
+      );
+      return;
+    }
+    this.labels.add(name);
+    console.log(this.labels);
+  }
+  // DEPRECATED: This method is deprecated in favor of .metric and .label
+  log(name, value) {
+    this.debugLog(
+      'context.iopipe.log is deprecated and will be removed in a future version, please use context.iopipe.metric'
+    );
+    this.metric(name, value);
```

Comment:
Is this technically a breaking change considering the new limits imposed?

Reply:
yes.

Reply:
Not considered a breaking change after investigation due to the fact that there was always a limit in the telemetry pipeline, it just was not exposed to agent documentation or functional implementation.